pkg/controllers/managementuserlegacy/istio/appHandler.go

```diff
@@ -1,159 +0,0 @@
-	cutils "github.com/rancher/rancher/pkg/catalog/utils"
-	v3 "github.com/rancher/rancher/pkg/generated/norman/project.cattle.io/v3"
-	"github.com/sirupsen/logrus"
-	"k8s.io/apimachinery/pkg/runtime"
-
-	mgmtv3 "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3"
-	apierrors "k8s.io/apimachinery/pkg/api/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-)
-
-const (
-	istioTemplateName = "rancher-istio"
-	istioAppName      = "cluster-istio"
-)
-
-var (
-	preDefinedIstioMetrics      = getPredefinedIstioMetrics()
-	preDefinedIstioClusterGraph = getPredefinedIstioClusterGraph()
-	PreDefinedIstioProjectGraph = getPredefinedIstioProjectGraph()
-)
-
-type appHandler struct {
-	istioClusterGraphClient  mgmtv3.ClusterMonitorGraphInterface
-	istioMonitorMetricClient mgmtv3.MonitorMetricInterface
-	clusterInterface         mgmtv3.ClusterInterface
-
-	clusterName string
-}
-
-func isIstioApp(obj *v3.App) bool {
-	if obj.Name != istioAppName {
-		return false
-	}
-	values, err := url.Parse(obj.Spec.ExternalID)
-	if err != nil {
-		logrus.Errorf("check catalog type failed: %s", err.Error())
-		return false
-	}
-
-	catalogWithNamespace := values.Query().Get("catalog")
-	split := strings.SplitN(catalogWithNamespace, "/", 2)
-
-	catalog := split[len(split)-1]
-
-	template := values.Query().Get("template")
-
-	if catalog == cutils.SystemLibraryName && template == istioTemplateName {
-		return true
-	}
-	return false
-}
-
-func isIstioMetricExpressionDeployed(clusterName string, istioClusterGraphClient mgmtv3.ClusterMonitorGraphInterface, istioMetricsClient mgmtv3.MonitorMetricInterface, obj *v3.App) error {
-	_, err := v32.IstioConditionMetricExpressionDeployed.DoUntilTrue(obj, func() (runtime.Object, error) {
-		for _, metric := range preDefinedIstioMetrics {
-			newObj := metric.DeepCopy()
-			newObj.Namespace = clusterName
-			if _, err := istioMetricsClient.Create(newObj); err != nil && !apierrors.IsAlreadyExists(err) {
-				return obj, err
-			}
-		}
-		for _, clusterGraph := range preDefinedIstioClusterGraph {
-			newObj := clusterGraph.DeepCopy()
-			newObj.Namespace = clusterName
-			if _, err := istioClusterGraphClient.Create(newObj); err != nil && !apierrors.IsAlreadyExists(err) {
-				return obj, err
-			}
-		}
-		for _, projectGraph := range PreDefinedIstioProjectGraph {
-			newObj := projectGraph.DeepCopy()
-			newObj.Namespace = clusterName
-			if _, err := istioClusterGraphClient.Create(newObj); err != nil && !apierrors.IsAlreadyExists(err) {
-				return obj, err
-			}
-		}
-
-		return obj, nil
-	})
-
-	if err != nil {
-		return errors.Wrapf(err, "failed to deploy istio metric expression into Cluster %s", clusterName)
-	}
-
-	logrus.Infof("deploy istio metric expression into CLuster %s successfully", clusterName)
-
-	return nil
-}
-
-func isIstioMetricExpressionWithdraw(clusterName string, istioClusterGraphClient mgmtv3.ClusterMonitorGraphInterface, istioMetricsClient mgmtv3.MonitorMetricInterface, obj *v3.App) error {
```

Comment:
if istio v1 is installed, should we trigger something to clean up the helm chart deployment on an upgrade?

Reply:
@aiyengar2 are we expecting users to manually reinstall the v2 istio chart in case of an upgrade? 
Simply triggering a cleanup might break things. Deleting configurations that were done by users as part of their istio setup could create issues.

Reply:
noted in https://github.com/rancher/rancher/pull/38682#issuecomment-1252541229